src/helpers/ExchangeRateRegistry.sol

```diff
@@ -0,0 +1,217 @@
+        );
+        transferOwnership(newOwner);
+        _initialized = true;
+    }
+
+    /**
+     * @notice Register new token to the registry
+     * @param token The token address to register
+     * @param pool The pool associated with the token
+     */
+    function registerToken(
+        address token,
+        address pool
+    ) external onlyOwner {
+        IBloomPool poolContract = IBloomPool(pool);
+        uint256 createdAt = poolContract.COMMIT_PHASE_END();
+
+        TokenInfo storage info = tokenInfos[token];
+        require(
+            !info.registered,
+            "ExchangeRateRegistry: token already registered"
+        );
+
+        info.registered = true;
+        info.active = true;
+        info.pool = pool;
+        info.createdAt = createdAt;
+
+        _activeTokens.add(token);
+
+        emit TokenRegistered(token, pool, createdAt);
+    }
+
+    /**
+     * @notice Activate the token
+     * @param token The token address to activate
+     */
+    function activateToken(address token) external onlyOwner {
+        TokenInfo storage info = tokenInfos[token];
+        require(info.registered, "ExchangeRateRegistry: token not registered");
+        require(!info.active, "ExchangeRateRegistry: token is active");
+
+        info.active = true;
+
+        _activeTokens.add(token);
+        _inactiveTokens.remove(token);
+
+        emit TokenActivated(token);
+    }
+
+    /**
+     * @notice Inactivate the token
+     * @param token The token address to inactivate
+     */
+    function inactivateToken(address token) external onlyOwner {
+        TokenInfo storage info = tokenInfos[token];
+        require(info.active, "ExchangeRateRegistry: token is inactive");
+
+        info.active = false;
+
+        _activeTokens.remove(token);
+        _inactiveTokens.add(token);
+
+        emit TokenInactivated(token);
+    }
+
+    /**
+     * @notice Return list of active tokens
+     */
+    function getActiveTokens() external view returns (address[] memory) {
+        return _activeTokens.values();
+    }
+
+    /**
+     * @notice Return list of inactive tokens
+     */
+    function getInactiveTokens() external view returns (address[] memory) {
+        return _inactiveTokens.values();
+    }
+
+    /**
+     * @notice Return true if the registry has been initialized
+     */
+    function isRegistryInitialized() external view returns (bool) {
+        return _initialized;
+    }
+
+    /**
+     * @notice Returns the current exchange rate of the given token
+     * @param token The token address
+     * @return The current exchange rate of the given token
+     */
+    function getExchangeRate(address token) external view returns (uint256) {
+        return _getExchangeRate(token);
+    }
+
+    function _getExchangeRate(address token) internal view returns (uint256) {
+        TokenInfo storage info = tokenInfos[token];
+        require(info.registered, "ExchangeRateRegistry: token not registered");
+
```

Comment:
Does it matter if token is also active?

Reply:
No we should still track the exchange rate of inactive tokens incase they are still in circulation on DEX's